javaparser-core/src/main/java/com/github/javaparser/printer/configuration/PrettyPrinterConfiguration.java

```diff
@@ -19,32 +19,30 @@
  * GNU Lesser General Public License for more details.
  */
 
-package com.github.javaparser.printer;
+package com.github.javaparser.printer.configuration;
 
-import static com.github.javaparser.utils.Utils.SYSTEM_EOL;
 import static com.github.javaparser.utils.Utils.assertNonNegative;
 import static com.github.javaparser.utils.Utils.assertNotNull;
 import static com.github.javaparser.utils.Utils.assertPositive;
 
-import java.util.function.Function;
+import java.util.Optional;
+import java.util.Set;
 
-import com.github.javaparser.ast.visitor.VoidVisitor;
-import com.github.javaparser.printer.configuration.Indentation;
+import com.github.javaparser.printer.PrettyPrinter;
 import com.github.javaparser.printer.configuration.Indentation.IndentType;
+import com.github.javaparser.printer.configuration.PrinterConfiguration.ConfigOption;
 
 /**
  * Configuration options for the {@link PrettyPrinter}.
+ * This class is no longer acceptable to use because it is not sufficiently configurable and it is too tied to a specific implementation
+ * <p> Use {@link ConfigurablePrinter interface or PrinterConfiguration default implementation } instead.
  */
-public class PrettyPrinterConfiguration {
+@Deprecated
+public class PrettyPrinterConfiguration implements ConfigurablePrinter {
+    
+    
+    ConfigurablePrinter wrapped;
```

Comment:
Please rename this to be more explicit - `wrappedConfiguration` for example (`wrapped` elsewhere refers to the wrapped node iirc)

Reply:
I do this asap.